src/Core/ServerSettings.h

```diff
@@ -50,6 +50,7 @@ namespace DB
     M(UInt64, max_temporary_data_on_disk_size, 0, "The maximum amount of storage that could be used for external aggregation, joins or sorting., ", 0) \
     M(String, temporary_data_in_cache, "", "Cache disk name for temporary data.", 0) \
     M(UInt64, aggregate_function_group_array_max_element_size, 0xFFFFFF, "Max array element size in bytes for groupArray function. This limit is checked at serialization and help to avoid large state size.", 0) \
+    M(Bool, aggregate_function_group_array_limit_size, false, "This is set to true. Array elements are truncated when they exceed the max array element size.", 0) \
```

Comment:
Not clear what this setting means. The name of the setting `aggregate_function_group_array_limit_size` tells us that it should contain some kind of size to be limited in groupArray function, but it is a bool. Also, not really clear what means `This is set to true.` when the default setting is set to false😕

Reply:
Thank you for your reply, I have renamed and added a new comment.